src/main/java/org/folio/rest/impl/GobiOrdersCustomMappingsImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.folio.rest.impl;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.gobi.HelperUtils.getEndpoint;
+import static org.folio.rest.core.RestClient.X_OKAPI_URL;
+
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.core.RestClient;
+import org.folio.rest.jaxrs.model.OrderMappings;
+import org.folio.rest.jaxrs.resource.GobiOrdersCustomMappings;
+import org.folio.rest.service.GobiCustomMappingsService;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+
+public class GobiOrdersCustomMappingsImpl extends BaseApi implements GobiOrdersCustomMappings {
+  private static final String GOBI_ORDERS_CUSTOM_MAPPINGS_LOCATION = getEndpoint(GobiOrdersCustomMappings.class) + "/%s";
+
+  @Override
+  public void getGobiOrdersCustomMappings(String query, int offset, int limit, String lang, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    RestClient rc = new RestClient(okapiHeaders, vertxContext);
+    new GobiCustomMappingsService(rc)
+      .getCustomMappingListByQuery(offset, limit)
+      .thenAccept(orderMappingsViewCollection -> asyncResultHandler.handle(succeededFuture(buildOkResponse(orderMappingsViewCollection))))
+      .exceptionally(fail -> handleErrorResponse(asyncResultHandler, fail));
```

Comment:
Can we do the same approach as in PostGobiOrdersHelper to create instances of RestClient and GobiCustomeMappingService in constructor?

Reply:
Its basically API implementation, so I am ok with current approach

Reply:
I was thinking vice versa, to pass RestClient as constructor parameter to PostGobiOrdersHelper. In other way it is harder to mock and test such constructed classes. I decided to leave it as is because it will require more refactoring. Perhaps it makes sense to do it in another story and refactor the code to leverage spring context in order to control life cycle of the objects